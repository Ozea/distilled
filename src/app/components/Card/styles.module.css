.card {
  box-shadow: 0 4px 12px #00000024;
  border-radius: 8px;
}