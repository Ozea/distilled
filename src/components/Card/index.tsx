import { PropsWithChildren } from 'react'
import { classNames } from '@/helpers/classNames'

import styles from './styles.module.css'

interface ICardProps {
  className?: string
}

export default function Card(props: PropsWithChildren<ICardProps>) {
  return <div className={classNames(styles.card, props.className)}>{props.children}</div>
}
