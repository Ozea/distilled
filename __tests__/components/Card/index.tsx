import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'

import Card from '@/components/Card'

describe('BorderingCountry', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Card />)
    expect(container).toMatchSnapshot()
  })
  it('should have correct original classname', () => {
    const { container } = render(<Card />)
    const div = container.getElementsByTagName('div')

    expect(div.item(0)?.className).toBe('card')
  })
  it('should have correct classnames', () => {
    const { container } = render(<Card className="test" />)
    const div = container.getElementsByTagName('div')

    expect(div.item(0)?.className).toBe('card test')
  })
  it('should render children correctly', () => {
    const headingText = 'Hello from Card child'
    const { container } = render(
      <Card>
        <h1>{headingText}</h1>
      </Card>
    )

    const div = container.getElementsByTagName('div')
    const heading = screen.getByRole('heading')

    expect(div.item(0)?.childElementCount).toBe(1)
    expect(heading).toHaveTextContent(headingText)
  })
})
