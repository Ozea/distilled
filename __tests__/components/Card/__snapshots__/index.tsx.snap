// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderingCountry matches the snapshot 1`] = `
<div>
  <div
    class="card"
  />
</div>
`;
